src/nvim/main.c

```diff
@@ -332,12 +332,6 @@ int main(int argc, char **argv)
 #endif
   bool use_builtin_ui = (has_term && !headless_mode && !embedded_mode && !silent_mode);
 
-  // don't bind the server yet, if we are using builtin ui.
-  // This will be done when nvim server has been forked from the ui process
-  if (!use_builtin_ui) {
-    server_init(params.listen_addr);
```

Comment:
Moved to after `ui_client_run`. That should be equivalent, right? Avoids the need for an explanatory comment and `!use_builtin_ui` condition.